tests/integration/components/o-s-s/nav-tab-test.ts

```diff
@@ -0,0 +1,122 @@
+      setupOnerror((err: any) => {
+        assert.equal(err.message, 'Assertion Failed: [component][OSS::NavTab] The parameter @tabArray is mandatory');
+      });
+      await render(hbs`<OSS::NavTab @onSelection={{this.onSelection}} />`);
+    });
+
+    test('It throws an error if @label and @icon is missing for any elment in @tabArray', async function (assert) {
+      this.tabArray.push({ label: '', icon: '' });
+      setupOnerror((err: any) => {
+        assert.equal(
+          err.message,
+          'Assertion Failed: [component][OSS::NavTab] The @label or @icon parameter is mandatory for each element in @tabArray'
+        );
+      });
+      await render(hbs`<OSS::NavTab @onSelection={{this.onSelection}} @tabArray={{this.tabArray}} />`);
+    });
+
+    test('Add icon in Tab', async function (assert) {
+      this.tabArray.push({ label: 'Tab', icon: 'far fa-thumbs-up' });
+      await render(hbs`<OSS::NavTab @tabArray={{this.tabArray}} @onSelection={{this.onSelection}} />`);
+
+      assert.dom('.tab-container').exists();
+      assert.dom('.tab-container .tab').exists();
+      assert.dom('.tab-content span').exists();
+      assert.dom('.tab-content i').exists();
+
+      assert.dom('.tab-content i').hasClass('fa-thumbs-up');
+      assert.dom('.tab-content span').hasText('Tab');
+    });
+
+    test('Add info-cirlce in Tab', async function (assert) {
+      this.tabArray.push({ label: 'Tab', infoCircle: true });
+      await render(hbs`<OSS::NavTab @tabArray={{this.tabArray}} @onSelection={{this.onSelection}} />`);
+
+      assert.dom('.tab-container').exists();
+      assert.dom('.tab-container .tab').exists();
+      assert.dom('.tab-content span').exists();
+      assert.dom('.tab-content i').exists();
+
+      assert.dom('.tab-content i').hasClass('fa-info-circle');
+      assert.dom('.tab-content span').hasText('Tab');
+    });
+
+    test('Add notification dot in Tab', async function (assert) {
+      this.tabArray.push({ label: 'Tab', notificationDot: true });
+      await render(hbs`<OSS::NavTab @tabArray={{this.tabArray}} @onSelection={{this.onSelection}} />`);
+
+      assert.dom('.tab-container').exists();
+      assert.dom('.tab-container .tab').exists();
+      assert.dom('.tab-content .fas.fa-circle').exists();
+    });
+
+    test('Add selected state on Tab', async function (assert) {
+      this.tabArray.push({ label: 'Tab', selected: true });
+      await render(hbs`<OSS::NavTab @tabArray={{this.tabArray}} @onSelection={{this.onSelection}} />`);
+
+      assert.dom('.tab-container').exists();
+      assert.dom('.tab-container .tab').exists();
+      assert.dom('.tab-content span').exists();
+
+      assert.dom('.tab-container .tab').hasClass('tab--selected');
+      assert.dom('.tab-container .tab .border-display').hasStyle({ borderBottom: '2px solid rgb(13, 13, 230)' });
+      assert.dom('.tab-content span').hasText('Tab');
+    });
+
+    test('Add disabled state on Tab', async function (assert) {
```

Comment:
`Tab displays disabled state properly`

Reply:
Also add a test that checks that the click event is not triggered on click.

- click on disabled tab
- assert that @onSelection has not been called